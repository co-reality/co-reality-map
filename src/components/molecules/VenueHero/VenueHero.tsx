import React from "react";

// Typings
import { DetailsPreviewProps } from "./VenueHero.types";

// Styles
import * as S from "./VenueHero.styles";
// import Button from "components/atoms/Button/Button";
import { Link } from "react-router-dom";
import { Button } from "react-bootstrap";

const VenueHero: React.FC<DetailsPreviewProps> = ({
  bannerImageUrl,
  logoImageUrl,
  name,
  subtitle,
  description,
  large = false,
  showEdit,
  venueId,
}) => {
  const renderLogo = () => <S.Logo backgroundImage={logoImageUrl} />;

  const renderName = () => (
    <S.TitleWrapper>
      <S.Title>{name}</S.Title>
      <S.Subtitle>{subtitle}</S.Subtitle>
    </S.TitleWrapper>
  );

  const renderDescription = () => <S.Description>{description}</S.Description>;

  const renderEditButton = () => (
    <Button as={Link} to={`/admin_v2/edit/${venueId}`}>
      Edit party info
    </Button>
  );

  return (
    <S.Container backgroundImage={bannerImageUrl} large={large}>
      {renderLogo()}
      {renderName()}
      {renderDescription()}
      {showEdit && renderEditButton()}
    </S.Container>
  );
};

export default VenueHero;
