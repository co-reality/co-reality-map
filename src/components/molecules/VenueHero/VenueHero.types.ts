export interface DetailsPreviewProps {
  bannerImageUrl?: string;
  logoImageUrl?: string;
  name?: string;
  subtitle?: string;
  description?: string;
  large?: boolean;
  venueId: string;
  showEdit?: boolean;
}
