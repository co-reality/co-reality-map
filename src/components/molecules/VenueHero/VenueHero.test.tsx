import React from "react";
import renderer from "react-test-renderer";

// Test component
import VenueHero from "./VenueHero";

describe("<VenueHero />", () => {
  test("renders correctly", () => {
    const tree = renderer.create(<VenueHero />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
