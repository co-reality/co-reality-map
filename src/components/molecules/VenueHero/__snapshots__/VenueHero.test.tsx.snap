// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<VenueHero /> renders correctly 1`] = `
<div
  className="sc-gsTCUz cRmWiU"
>
  <div
    className="sc-jSgupP fnAQnX"
  />
  <div
    className="sc-dlfnbm cxRrbB"
  >
    <h1
      className="sc-hKgILt kLrxSF"
    />
    <h2
      className="sc-eCssSg fRLehA"
    />
  </div>
  <p
    className="sc-gKsewC dravmq"
  />
</div>
`;
